contracts/Accounts.sol

```diff
@@ -2,15 +2,26 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
-
+interface TokenInterface {
+  function signUpReward(address userAddress,uint256 amount) external ;
+}
 // Matic Mumbai: 0xB4341D98e12B6e93dcf231dd615B16d8b208C83c
 contract Accounts is ERC721URIStorage {
     
-    constructor() ERC721("Game Profile Token","GPT"){}
+    TokenInterface TI;
+    
+    constructor() ERC721("Game Profile Token","GPT"){
+        
+    }
     
     function signUp(string memory accountURI, uint256 accountID) external{
         _safeMint(_msgSender(), accountID);
         _setTokenURI(accountID, accountURI);
+        TI.signUpReward(msg.sender, 100);
+    }
+    
+    function AddTIAddress(address tokenAddress) public {
+        TI = TokenInterface(tokenAddress);
```

Comment:
This shouldn't be public.

Reply:
It has to be public but I'll change it to onlyOwner. We need to set the contract address after all the contracts have been deployed therefore this.